reconstruction/ecoli/knowledge_base_raw.py

```diff
@@ -371,14 +373,16 @@ def _join_data(self):
 				added_data = getattr(added_data, attr)
 
 			# Check columns are the same for each dataset
-			col_diff = set(data[0].keys()).symmetric_difference(added_data[0].keys())
-			if col_diff:
-				raise ValueError(f'Could not join datasets {data_attr} and {attr_to_add} '
-					f'because columns do not match (different columns: {col_diff}).')
-
-			# Join datasets
-			for row in added_data:
-				data.append(row)
+			if added_data:
```

Comment:
I don't think this change would be necessary if we remove the empty file requirements for the metabolism files?